managed/src/main/java/com/yugabyte/yw/models/Region.java

```diff
@@ -86,52 +85,57 @@ public void setActiveFlag(Boolean active) {
     this.active = active;
   }
 
+  class RegionDetails {
+    public String sg_id; // Security group ID.
+    public String vet; // Vnet key.
+  }
+
   @DbJson
   @Column(columnDefinition = "TEXT")
-  public JsonNode details;
+  public RegionDetails details;
 
   public void setSecurityGroupId(String securityGroupId) {
     if (details == null) {
-      details = Json.newObject();
+      details = new RegionDetails();
     }
-    ((ObjectNode) details).put(SECURITY_GROUP_KEY, securityGroupId);
+    details.sg_id = securityGroupId;
     save();
   }
 
   public String getSecurityGroupId() {
     if (details != null) {
-      JsonNode sgNode = details.get(SECURITY_GROUP_KEY);
-      return sgNode == null || sgNode.isNull() ? null : sgNode.asText();
+      String sgNode = details.sg_id;
+      return sgNode == null || sgNode.isEmpty() ? null : sgNode;
     }
     return null;
   }
 
   public void setVnetName(String vnetName) {
     if (details == null) {
-      details = Json.newObject();
+      details = new RegionDetails();
     }
-    ((ObjectNode) details).put(VNET_KEY, vnetName);
+    details.vet = vnetName;
     save();
   }
 
   public String getVnetName() {
     if (details != null) {
-      JsonNode vnetNode = details.get(VNET_KEY);
-      return vnetNode == null || vnetNode.isNull() ? null : vnetNode.asText();
+      String vnetNode = details.vet;
+      return vnetNode == null || vnetNode.isEmpty() ? null : vnetNode;
     }
     return null;
   }
 
   @DbJson
   @Column(columnDefinition = "TEXT")
-  public JsonNode config;
+  public Map<String, String> config;
 
   public void setConfig(Map<String, String> configMap) {
```

Comment:
Tyr to capture as many keys of this map as possible in static consts. 
For example I see "OVERRIDES" for kubernetes. There may be more.

Reply:
are you suggesting to use POJO here?

Reply:
If you can figure out all the fields yes. But I suspect that it will merge in random things like some gcp service account creds etc. i.e. it is inherantly open-ended.

I am just saying that capture in static constants at least the keys that we know of. You can do that in follow up. This is similar to `VNET_KEY` and `SECURITY_GROUP_KEY` in case of details. It at least give some sanity to code as we can track references and most keys get tracked at one place.